File: Codes/3074-apple-redistribution-into-boxes.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-12-24 13:30:54
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-12-24 13:32:26
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        sort(capacity.begin(), capacity.end(), greater<>());
        int cnt = 0;
        for (int t : apple) {
            cnt += t;
        }
        int ans = 0;
        for (int t : capacity) {
            cnt -= t;
            ans++;
            if (cnt <= 0) {
                return ans;
            }
        }
        return -1;  // Fake Return
    }
};

File: Codes/3074-apple-redistribution-into-boxes.go
/*
 * @Author: LetMeFly
 * @Date: 2025-12-24 13:30:54
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-12-24 13:45:02
 */
package main

import "sort"

func minimumBoxes(apple []int, capacity []int) int {
    cnt := 0
    for _, t := range apple {
        cnt += t
    }
    sort.Sort(sort.Reverse(sort.IntSlice(capacity)))
    for i, t := range capacity {
        cnt -= t
        if cnt <= 0 {
            return i + 1
        }
    }
    return -1  // Fake Return
}